scss/numerictextbox/_theme.scss

```diff
@@ -40,4 +42,16 @@
 
     }
 
+    .k-numeric-wrap.k-state-invalid {
+        border-color: $input-validation-error;
+
+        .k-i-warning {
+            color: $input-validation-error;
+        }
+
+        input {
+            color: $input-validation-error;
```

Comment:
use `color: inherit` in base file (if not already used) and set the color to the wrapping element.